src/main/java/com/example/demo/config/KafkaConfig.java

```diff
@@ -0,0 +1,44 @@
+package com.example.demo.config;
+
+import com.example.demo.entity.UserEntity;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.annotation.EnableKafka;
+import org.springframework.kafka.core.DefaultKafkaProducerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.core.ProducerFactory;
+import org.springframework.kafka.support.serializer.JsonSerializer;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Configuration
+@EnableKafka
+public class KafkaConfig {
+
+    @Value("${spring.kafka.bootstrap-servers}")
+    private String bootstrapServers;
+
+    @Bean
+    public ProducerFactory<String, String> producerFactory() {
+        Map<String, Object> configProps = new HashMap<>();
+        configProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+        configProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+        configProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, JsonSerializer.class);
+        return new DefaultKafkaProducerFactory<>(configProps);
+    }
+
+    @Bean
+    public KafkaTemplate<String, String> kafkaTemplate() {
+        return new KafkaTemplate<>(producerFactory());
+    }
+
+    @Bean
+    public NewTopic userRegistrationTopic() {
+        return new NewTopic("userTopic", 1, (short) 1);
+    }
```

Comment:
partition sayısını arttırmak aynı zamanda birden fazla mesajı okuyabilmeni sağlar, bilgi olsun diye yazıyorum.
Fakat dikkatli kullanılması gerek. Çünkü her partition sırayı garanti eder fakat aynı mesaj zinciri farklı partitionlara dağıldıysa mesaj sırası garanti edilmez. Mesaj sırasını korumak için kafkada key kullanılır. Örnek senaryo:
Kullanıcı sipariş oluşturdu, hemen peşine kargo oluştu daha sonra mail gitti ve en son da sms. Bunların hepsini kafka topicleri ile yapıldığını düşün. Eğer mesajlara key verilmezse sipariş oluşmadan önce kargo oluşabilir. Key verilirse hepsi aynı topiğin aynı partitionuna gideceğini garanti ettiği için sıra korunur. fyi